conans/client/build/meson.py

```diff
@@ -118,6 +118,44 @@ def _get_dirs(self, source_folder, build_folder, source_dir, build_dir, cache_bu
     def flags(self):
         return defs_to_string(self.options)
 
+    def _configure_cross_compile(self, cross_filename, environ_append):
+        cpu_translate = {
+            'armv8': ('aarch64', 'aarch64', 'little'),
+            'x86': ('x86', 'x86', 'little'),
+            'x86_64': ('x86_64', 'x86_64', 'little'),
+        }
+        cpu_family, cpu, endian = cpu_translate[str(self._conanfile.settings.arch)]
+        cflags = ', '.join(repr(x) for x in os.environ.get('CFLAGS', '').split(' '))
+        cc = os.environ.get('CC', 'cc')
+        cpp = os.environ.get('CXX', 'c++')
+        with open(cross_filename, "w") as fd:
+            fd.write("""
+                [host_machine]
+                system = '{os}'
+                cpu_family = '{cpu_family}'
+                cpu = '{cpu}'
+                endian = '{endian}'
+
+                [properties]
+                needs_exe_wrapper = true
+                c_args = [{cflags}]
+                [binaries]
+                c = '{cc}'
+                cpp = '{cpp}'
```

Comment:
we need to respect other tools if corresponding environment variables are set (e.g. AR, RANLIB, STRIP, etc.)